Tools/ML/ONNX/model.cxx

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file     model.cxx
+///
+/// \author   Christian Sonnabend <christian.sonnabend@cern.ch>
+///
+/// \brief    A general-purpose class with functions for ONNX model applications
+///
+
+// ONNX includes
+#include "Tools/ML/ONNX/model.h"
+
+namespace o2
+{
+
+namespace ml
+{
+
+std::string OnnxModel::printShape(const std::vector<int64_t>& v)
+{
+  std::stringstream ss("");
+  for (size_t i = 0; i < v.size() - 1; i++)
+    ss << v[i] << "x";
+  ss << v[v.size() - 1];
+  return ss.str();
+}
+
+OnnxModel& OnnxModel::operator=(OnnxModel& inst)
+{
+
+  mEnv = inst.mEnv;
+  mSession = inst.mSession;
+
+  mInputNames = inst.mInputNames;
+  mInputShapes = inst.mInputShapes;
+  mOutputNames = inst.mOutputNames;
+  mOutputShapes = inst.mOutputShapes;
+
+  modelPath = inst.modelPath;
+  ccdbUrl = inst.ccdbUrl;
+  activeThreads = inst.activeThreads;
+  valid_from = inst.valid_from;
+  valid_until = inst.valid_until;
+
+  LOG(debug) << "Model copied!";
```

Comment:
ok, so you assume a copy of an instance doesn't have the same session options?

Reply:
Technically, the = operator does not exist for two models in ONNX (I think it is explicitly deleted). That's why I created a new one here. Maybe there is some better way of doing it, but for now this should work. Of course this should be extended in the future!

Reply:
If operator= is deleted, then maybe there is a good reason in ONNX. Performance or logic-wise. Where do you need it?

Reply:
Indeed, I don't see good reasons for copying the model. I would remove the = operator from the class.